main.go

```diff
@@ -24,18 +23,34 @@ func HelloUser(resp http.ResponseWriter, req *http.Request) {
 
 func CheckUserID(resp http.ResponseWriter, req *http.Request) {
 	id := req.URL.Query().Get("userid")
-	if isUserIdValid(id) {
+	if isUserIDValid(id) {
 		fmt.Fprintln(resp, "Hello, user!")
 	} else {
 		http.Error(resp, "Bad user ID", http.StatusForbidden)
 	}
 }
 
-func isUserIdValid(id string) bool {
-	for _, validUser := range validUserIDs {
-		if id == validUser {
-			return true
-		}
+func AddUser(resp http.ResponseWriter, req *http.Request) {
+	id := req.URL.Query().Get("id")
+	if id != "" {
+		validUserIDs.Store(id, true)
+		fmt.Fprintf(resp, "User %s added successfully!\n", id)
+	} else {
+		http.Error(resp, "Invalid user ID", http.StatusBadRequest)
+	}
+}
+
+func DeleteUser(resp http.ResponseWriter, req *http.Request) {
+	id := req.URL.Query().Get("id")
+	if id != "" {
+		validUserIDs.Delete(id)
+		fmt.Fprintf(resp, "User %s deleted successfully!\n", id)
+	} else {
+		http.Error(resp, "Invalid user ID", http.StatusBadRequest)
```

Comment:
Could you improve this error msg by saying something about the missing `id` param?